### PR TITLE
fix sns event deserialize issue

### PR DESCRIPTION
**Pull request by bnusunny**
SNSEvent, CodeCommitEvent and ScheduledEvent used Joda DateTime. Added JodaModule to Jackson, so that these events can be properly deserialized. 

aws-lambda-java-events uses old Joda-time v2.6, which conflict with JodaModule. I excluded joda-time from aws-lambda-java-events and used the latest version from JodaModule. 

Solves issue #6491

### CODE REVIEW

**Comment by machi1990** [comment]
@geoand I asked you for a review since it seems jackson related :-)

**Comment by geoand** [comment]
@machi1990 thanks. I'll check in a couple hours

**Review by evanchooly — APPROVED** [review]
looks pretty straightforward to me.

File: extensions/amazon-lambda/runtime/pom.xml [hunk 1]
```diff
@@ -23,6 +23,13 @@
         <dependency>
             <groupId>com.amazonaws</groupId>
             <artifactId>aws-lambda-java-events</artifactId>
+            <version>2.2.7</version>
```
Comment (geoand) [inline extensions/amazon-lambda/runtime/pom.xml]: We shouldn't be overriding the version here, but instead change it here: https://github.com/quarkusio/quarkus/blob/master/bom/runtime/pom.xml#L139

File: extensions/amazon-lambda/runtime/src/main/java/io/quarkus/amazon/lambda/runtime/AmazonLambdaRecorder.java [hunk 2]
```diff
@@ -49,7 +50,8 @@ public void setHandlerClass(Class<? extends RequestHandler<?, ?>> handler, BeanC
         beanContainer = container;
         AmazonLambdaRecorder.objectMapper = getObjectMapper()
                 .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
-                .configure(MapperFeature.ACCEPT_CASE_INSENSITIVE_PROPERTIES, true);
+                .configure(MapperFeature.ACCEPT_CASE_INSENSITIVE_PROPERTIES, true)
+                .registerModule(new JodaModule());
```
Comment (geoand) [inline extensions/amazon-lambda/runtime/src/main/java/io/quarkus/amazon/lambda/runtime/AmazonLambdaRecorder.java]: I let's go with this for now, but I think we'll want to introduce an `ObjectMapperCustomizer` to the module that takes care of all this

**Review by geoand — CHANGES_REQUESTED** [review]
I added one comment for another time and one request for a change

**Comment by patriot1burke** [comment]
@geoand I'm working on closing out some of the lambda bugs.  I'll add customizer capability to the list.

**Comment by geoand** [comment]
Cool, thanks @patriot1burke

File: extensions/amazon-lambda/runtime/pom.xml (hunk 1 above)
Comment (bnusunny) [reply extensions/amazon-lambda/runtime/pom.xml]: Sure. Updated.

**Review by geoand — APPROVED** [review]

**Pull request merged**